dapp/src/components/name-card/ExtendedNameCard.tsx

```diff
@@ -46,17 +46,24 @@ export function ExtendedNameCard({
 
     const targetAddress = nameRecord?.nameRecord?.targetAddress;
     const expired = nameRecord?.nameRecord ? isNameRecordExpired(nameRecord.nameRecord) : false;
-
+    const isNameSubname = nameRecord?.nameRecord ? isSubname(nameRecord.nameRecord.name) : false;
+    const isNameGracePeriodExpired = isGracePeriodExpired(nft);
     const buttonText = (() => {
-        if (expired) {
+        if (expired && !isNameGracePeriodExpired) {
             return 'Renew Name';
         }
+        if (isNameGracePeriodExpired) {
+            return isNameSubname ? 'Delete Subname' : 'Delete Name';
+        }
         return targetAddress ? formatAddress(targetAddress) : 'Connect to address';
     })();
 
     const handleButtonClick = () => {
-        if (expired) {
+        if (expired && !isNameGracePeriodExpired) {
             openDialog(NameDialogId.RenewName);
+        }
```

Comment:
Otherwise the `ConnectToAddress` will open.
```suggestion
        } else if (isNameGracePeriodExpired) {
```